### ISSUE TITLE
FNX-5486 ⁃ Enable selection of what to sync

### ISSUE THREAD

**Issue by sv-ohorvath** [mozilla-mobile/fenix#2157]
### User Story
As a user, I can choose what data I want to sync across my Firefox connected devices, so I have control of my data and what is shared between devices.

### Acceptance Criteria
-I can go into my sync settings and disable/enable history sync
-I can go into my sync settings and disable/enable bookmarks sync
-I can choose whether I want to sync history/bookmarks/etc before sync begins (@ryanfeeley do we have Mocks for this?)

https://mozilla.invisionapp.com/share/WTT2CXNDUN8#/374762784_Main_Browser_-_Settings

Dependencies tracked on this board: https://github.com/orgs/mozilla-mobile/projects/37

**Comment by grigoryk** [comment]
This needs a-c and a-s support: https://github.com/mozilla-mobile/android-components/issues/1584

**Comment by liuche** [comment]
@lime124 I know we mentioned earlier that this doesn't require UX because we can just "turn on" the sync settings, but there is a piece that we should do, which is allow users BEFORE they start syncing to choose what they want to sync. More context here in [this doc](https://docs.google.com/document/d/1jiaVYI1i1JS31ZEKjbzLzexK5TJBrYKO58dcl5wynwI/edit#heading=h.o4u07u901u81).

**Comment by vesta0** [comment]
@liuche updated the acceptance criteria and moved this to prioritized UX backlog.

**Comment by lime124** [comment]
@liuche @vesta0 thanks to michelle i found the spec! can you take a look and see if this covers everything you need? 

quickly looking it over, the only thing that isn't clear to me is how it comes up after you sign in with your account. if there's more design work needed, let's loop in @apbitner as i think the most frequent case will be coming off onboarding flow.

**Comment by liuche** [comment]
@lime124 do I need a Mozilla invision account? All the past invision links have been public but I can't see this one without signing in, but I don't think I have a Mozilla account for Invision.

**Comment by lime124** [comment]
whoops sorry about that - i pasted the URL instead of the share link. it should work now.

**Comment by ryanfeeley** [comment]
@lime124 My mockup linked above shows the desired experience (switches beside History and Bookmarks). What was shipped was a radically descoped version that did not allow the user to interact with these selections, nor did it even accurately indicate the users preference at the account level (meaning if they had disabled history syncing, the read-only checkmark did not reflect that).

If I understand this issue correctly, the aim is to do the engineering to make the feature work as designed.

**Comment by ryanfeeley** [comment]
>  I can choose whether I want to sync history/bookmarks/etc before sync begins (@ryanfeeley do we have Mocks for this?)

This would be handled on the web using our "choose what to sync" screen, which @vladikoff can speak to. If the user has never consented to syncing at the account level, they should see this screen when signing in to continue to Sync (and always see it when creating an account to continue to Sync).

**Comment by ekager** [comment]
According to @grigoryk 's comment above I think this is still waiting on AC mozilla-mobile/android-components#2727

**Comment by grigoryk** [comment]
Pre-req: https://github.com/mozilla-mobile/fenix/pull/4931

**Comment by sblatz** [comment]
This is still [waiting on the AC implementation](https://github.com/mozilla-mobile/android-components/issues/2727)

**Comment by sblatz** [comment]
@ekager will touch base with @grigoryk to see if he's doing all of the fenix work for this or not. 

If he is, @ekager please remove the point estimate here.
